chapters/annotations.tex

```diff
@@ -817,9 +817,17 @@ \subsubsection{Text}\doublelabel{text}
 outline). The contents inherited from \lstinline!FilledShape! is deprecated, but kept for compatibility reasons.
 
 There are a number of common macros that can be used in the text, and
-they should be replaced when displaying the text as follows:
+they should be replaced when displaying the text as follows (in order such that the earliest ones have precedence,
+and using the longest sequence of identifier characters - alphanumeric and \_):
```

Comment:
Shouldn't the rule of maximum munge only be applied after checking for the special cases at the top of the list?  I may not be a big deal, but given a parameter called `names`, you could then do `%{names}` to refer to the parameter, and use `%names` to refer to the name immediately followed by 's'.  For a parameter called `p`, we have solved this sort of problem by having the preferred from `%{p}s`, and I think it would be nice to have the same expressiveness also for the special cases for the sake of completeness.

Reply:
To me that just looks odd, for values I could see cases such as 2x3 (from `%{w}x%{h}`), but for class and component-name I don't see a real good use case. 

The case `%names` suggest plural in English and that will not work reliably: it works for inertia - but not for clutch (as the plural is clutches). I believe that could be considered in the internationalization of libraries

Reply:
Yes, that's why it may not be a big deal.  It just strikes me as some kind of design error when certain things can't be expressed.